ParametricObjectCreation/HoneycombSolid/HoneycombSolid.py

```diff
@@ -0,0 +1,235 @@
+		obj.addProperty("App::PropertyLength","Height","Honeycomb", "Height of the Honeycomb").Height=2.0
+		obj.addProperty("App::PropertyLength","Circumradius","Polygon","Radius of the inner circle").Circumradius=5.0
+        	#obj.addProperty("App::PropertyLength","Edges","Polygon","Poligon number of edges").Edges=6.0
+		obj.addProperty("App::PropertyLength","Tickness","Walls","Tickness of the honeycomb walls").Tickness=1.0
+		obj.Proxy = self
+	
+		self.Type = 'HoneycombSolid'
+		self.Version = __version__
+        
+
+	def onChanged(self, fp, prop):
+		''' Print the name of the property that has changed '''
+		FreeCAD.Console.PrintMessage("Change property: " + str(prop) + "\n")
+
+	def execute(self, fp):
+		''' Print a short message when doing a recomputation, this method is mandatory '''
+		FreeCAD.Console.PrintMessage("Recompute Python HoneycombSolid feature\n")
+
+		length = fp.Length
+		width = fp.Width
+		height = fp.Height
+		radius = fp.Circumradius
+		tickness = fp.Tickness
+        
+		#edges = fp.Edges
+		edges = 6
+
+		#######################################################################
+		# Container box, used to cut the polygon array
+		container = Part.makeBox(length,width,height)
+
+		#######################################################################
+		# create the first polygon
+		figure = []
+		m=Base.Matrix()
+		edges_angle = math.radians(360.0/edges)
+		m.rotateZ(edges_angle)
+		v=Base.Vector(radius,0,0)
+
+		for i in range(edges):
+			figure.append(v)
+			v = m.multiply(v)
+
+		figure.append(v)
+		polygon = Part.makePolygon(figure)
+
+		# move it to the center of the container box	
+		polygon.translate(Base.Vector(length/2, width/2, 0))
+
+		# create a face for the first polygon
+		f1=Part.Face([polygon])
+
+		#######################################################################
+		# create copies of poligon using radial pattern
+
+		# calculate how many circunferencias needs to cover the maximun length of the container box
+		n_cols = math.ceil( (length / (radius + tickness) / 2) )
+		n_rows = math.ceil( (width / (radius + tickness) / 2) + 3)
+		# FreeCAD.Console.PrintMessage("n_cols: " + str(n_cols) + " n_rows: " + str(n_rows) + "\n")
+	
+		# To store all the poligons face
+		e_faces = []
+		e_faces.append(f1) # add the first one created before
+
+		# Iterate over each imaginary circle which circunference contains the center of the polygon circle 
+		for column in range(-n_cols, n_cols):
+			for row in range(-n_rows, n_rows):
+				# omit the first
+				if (column != 0) or (row != 0):
+					if column % 2 != 0:
+					# odd
+						if row % 2 != 0:
+						# odd
+							centers_distance = tickness + 2 * radius * math.sin(edges_angle)
+							x_delta = centers_distance *  math.sin(edges_angle)
+							y_delta = centers_distance * math.cos(edges_angle)
+
+							x_origin = column * x_delta 
+							y_origin = row * y_delta
+		
+							delta_x_y = Base.Vector(x_origin, y_origin,0)
```

Comment:
Lines 97 to 104 could be taken out-of the `if` statement because they repeat at line 114 (assuming you accept my previous comment). I think there are other lines that also repeat and could be place outside the `if`, in the case that it's ok for you to change the order in which the variables are computed.

Reply:
@galou, to be honest, I don't understand too much how to apply the improvement you are recommending.

I would like to expend more time to see if I get a better idea from your recommendation.

Maybe you can create a new PR if you know exactly how to improve that, will be great!